packages/pug-code-gen/index.js

```diff
@@ -97,15 +110,64 @@ Compiler.prototype = {
     throw err;
   },
 
+  parseExpr: function(expr) {
+    //return babylon.parse('function*g(){return e='+expr+'}').program.body[0].body.body[0].argument.right;
+    return babylon.parseExpression(expr);
+  },
+  parseArgs: function(args) {
+    return babylon.parse("a("+args+")").program.body.pop().expression.arguments;
+  },
+  btpl_addWith: function() {
+    var globals = this.options.globals ? this.options.globals.concat(INTERNAL_VARIABLES) : INTERNAL_VARIABLES;
+    globals.concat(this.runtimeFunctionsUsed.map(function (name) { return 'pug_' + name; }));
+    var tpl = "// @with exclude: "+ globals.join(",") +"\n{\nlocals || {};\nSOURCE;\n}"
+    var tplc = babelTemplate(tpl, { preserveComments: true });
+    return tplc;
+  },
+  ast_variableDeclaration: function() {
+    return t.variableDeclaration('var', [
+          t.variableDeclarator(t.identifier('pug_html'), t.stringLiteral('')),
+          t.variableDeclarator(t.identifier('pug_mixins'), t.objectExpression([])),
+          t.variableDeclarator(t.identifier('pug_interp'), null)
+        ])
+  },
+  ast_return: function() {
+    return [t.emptyStatement(), t.returnStatement(t.identifier('pug_html'))];
```

Comment:
This shouldn't need the `emptyStatement`. The `;` was there originally because I wasn't confident that the code for the template would have been ended in `;` or newline. Since it's all a babel ast, we can drop it now.

Reply:
yes i was trying to reach source parity when i developed this. The point was to have all the tests pass.